mm2src/mm2_main/src/lp_swap/taker_swap_v2.rs

```diff
@@ -817,7 +828,11 @@ impl<MakerCoin: MmCoin + MakerCoinSwapOpsV2, TakerCoin: MmCoin + TakerCoinSwapOp
                     .push(new_locked);
             },
             TakerSwapEvent::TakerFundingSent { .. } => {
-                let swaps_ctx = SwapsContext::from_ctx(&self.ctx).expect("from_ctx should not fail at this point");
+                let swaps_ctx = SwapsContext::from_ctx(
+                    &self.ctx,
+                    self.taker_coin.account_db_id().expect("Valid maker pubkey").as_deref(),
```

Comment:
This kills the entire purpose of using the `Option`al result type for `account_db_id`. The idea behind using the `Option` result type for `account_db_id` was to fallback to mm2 internal pubkey for cases where `account_db_id` returns `None` (and it should return None unless there is a coin specific `account_db_id` implementation (like implementing it in `impl MmCoin for TendermintCoin`)). This implies that we should refrain from `unwrap`ping or `expect`ing the `Option`. Instead, it should be used or passed as is, and in the leaf function, use `unwrap_or_else(|| internal_pubkey_value)` to handle the fallback scenario.